### ISSUE TITLE
Running Error

### ISSUE THREAD

**Issue by canhe173** [tlpss/keypoint-detection#14]
Hello, I met a question when I run the test code `bash test/integration_test.sh`.  The error was as follows:

![Error log](https://user-images.githubusercontent.com/97797451/205898631-b24685d6-73c9-48ad-b082-7f95326b7823.png)

What's more, a similar problem occurred when I used my own annotated datasets.

![owner codes](https://user-images.githubusercontent.com/97797451/205900422-0f988d2c-f5af-40d5-b8f8-c9ff2a1d378b.png)

![owner error log](https://user-images.githubusercontent.com/97797451/205900290-0d3aeea9-faa6-41e7-ab58-70f5c04efbcb.png)

**Comment by tlpss** [comment]
Hi @canhe173  could you provide some more information on your system? maybe a list of the packages in  your conda and/or pip environment?

 And what shell are you using?

**Comment by canhe173** [comment]
> 

Thanks for your reply, My environment:
- OS: [Ubuntu 18.04]
- PyTorch Version [1.13.0]
- torchvision version [0.14.0]
- cudatoolkit version [11.3.1]
- Nvidia driven version [470.161.03]
- GPU [GeForce RTX 3080Ti]
When I run `bash test/integration_test.sh` in vs code's terminal window, Ubuntu's own terminal window, I get similar errors.

**Comment by tlpss** [comment]
Hi @canhe173 , I have done some experiments myself and I think that you might have extra spaces after the line split `\ ` in the bash script, which results in a similar error. So make sure that each `\` is immediately followed by a Return without additional spaces. 

If you're not sure what I mean, just upload your bash file and I can have a look.

Let me know whether this fixed the issues you have.

**Comment by canhe173** [comment]
Hi @tlpss, thanks for your  timely reply, the reason why I found the previous problem is that I have comments between bash commands. I deleted the comments to solve that problem.

However, I found that codes could not handle the keypoints of multi categories. For example, If I have two categories of keypoints, the program will report errors halfway. When I read the keypoints of the datasets with `visualization.py`, I found that it could not obtain all the keypoints, as shown in the following figures.The sixth image should have six keypoints, but it only reads the first four keypoints. The list of keypoints should be divided into two categories: towels and clothes, but the code seems to read both into one category.

![result figure](https://user-images.githubusercontent.com/97797451/206650651-320d37a2-162d-4925-89d1-179cfdfd2acc.png)
![results](https://user-images.githubusercontent.com/97797451/206650765-8e0e1df4-fc88-4234-afa4-cb475cdd2396.png)
![class](https://user-images.githubusercontent.com/97797451/206650974-7d1ca64c-bf49-427b-bc07-7b169712d4c0.png)

**Comment by tlpss** [comment]
Hi @canhe173, glad you managed to resolve the first issue!

Looking at the screenshots you provided, it would indeed seem that something is wrong with the keypoint to channel assignment. Could you send me your coco dataset (or a part of it, at least containing annotations for both types?), that would make it easier for me to find out what is going on? I'll make sure to look into it very soon

**Comment by canhe173** [comment]
Hi @tlpss, I have solved that problem because there is a bug in program `coco_dataset.py`. Changing `category = category_dict[category.id]` in 128 lines to `category = category_dict[annotation.category_id]` can solve this problem. Directly using `category.id` cannot obtain the correct ID.

**Comment by tlpss** [comment]
@canhe173, you are right :+1: Thanks for catching that bug! If you have a Pull Request for it I would be happy to review and merge, if not I will add the fix myself

**Comment by canhe173** [comment]
Hi @tlpss, I have pulled request for that bug, you can review it. I would also like to ask you how to call the trained model for testing because I could not find that information in the models folder.

**Comment by tlpss** [comment]
@canhe173, thanks a lot! I'll finish merging it (and change the integration test to include mutliple categories). 

>  I would also like to ask you how to call the trained model for testing because I could not find that information in the models folder.

There is no code in this repo for inference atm, but you can find an example [here](https://github.com/Victorlouisdg/iros-2022-cloth-competition/blob/main/cloth-manipulation/cloth_manipulation/detector.py) 

Easiest flow (imo) is going to weights and biases, getting the reference of the checkpoint you would like to use and then creating a model from that checkpoint. Don't forget to set your model to eval with `model.eval()` , that is something I tend to forget often..

I'll try to add an example to this repo as well.
